### PR TITLE
DEVELOP-2147: Add support for Miseq Nano and Micro

### PR DESCRIPTION
**Pull request by matrulda**
This PR adds support for special MiSeq flowcells (nano and micro) that generates less data than the standard one.

@mikaelamo It would be nice if you could review the changes in config.yaml since you are familiar with INS-00123. :)

### CODE REVIEW

File: checkQC/run_type_recognizer.py [hunk 1]
```diff
@@ -101,16 +101,36 @@ def name():
...
+
+        def _flowcell_type(runtype_recognizer):
+            try:
+                tiles_per_swath = int(runtype_recognizer.run_parameters["RunParameters"]["Setup"]["NumTilesPerSwath"])
+                if tiles_per_swath == 2:
+                    return "nano"
+                elif tiles_per_swath == 4:
+                    return "micro"
+                elif tiles_per_swath >= 14:
+                    return "standard"
```
Comment (Aratz) [inline checkQC/run_type_recognizer.py:124]: Is it possible that `tiles_per_swath` does not match any of these values? In this case maybe you could raise the `ReagentVersionUnknown` exception?

**Review by Aratz — APPROVED** [review]
Looks good to me :+1: I just left one question in the code.

**Review by mikaelamo — APPROVED** [review]
Looks correct in regards to INS-00123.

File: checkQC/run_type_recognizer.py (hunk 1 above)
Comment (matrulda) [reply checkQC/run_type_recognizer.py:124]: Ah good catch, I think it shouldn't be possible with the current MiSeq configurations, but Illumina might add something new in the future. Will fix!

**Comment by matrulda** [comment]
Thanks @mikaelamo and @Aratz for the reviews :pray: ! @Aratz if you could just check the last commit to make sure I didn't introduce anything weird, I think we are good to go.

**Comment by Aratz** [comment]
Looks good, great job :+1:

**Comment by Aratz** [comment]
Also I just noticed Python is now introducing [match statements](https://docs.python.org/3/tutorial/controlflow.html?highlight=match%20case#match-statements) in version 3.10, so I guess in the future this will become the canonical way of writing this kind of code.

**Review by mikaelamo — APPROVED** [review]

**Comment by matrulda** [comment]
Good input @Aratz, I'm not sure how to handle the `>= 14` condition using match case though. :thinking: Do you have an idea? Otherwise I think I'll keep it like this.

**Comment by Aratz** [comment]
I'd write `case x if x >= 14:`, but I think it's good as it is now, we don't use Python 3.10 yet do we?

**Comment by matrulda** [comment]
Ah ofc, thanks, @Aratz ! I went ahead and upgraded to py3.10 since it required very few changes.

**Comment by matrulda** [comment]
Hmm, I had to update the ubuntu version for travis to be able to download py3.10. We should make sure that we can install py3.10 on the biotanks before including those changes in the next release.

**Review by Aratz — APPROVED** [review]
Looks good to me :+1:

**Comment by matrulda** [comment]
I removed the py3.10 commits, our old servers can't handle py3.10. Will add those changes in a separate PR so that we can release these changes first.

**Pull request merged**